plugins/generator-1.19.2/forge-1.19.2/templates/json/item.json.ftl

```diff
@@ -4,7 +4,22 @@
         <#if var_item??>
             "layer0": "${modid}:items/${data.getItemTextureFor(var_item)}"
         <#else>
-            "layer0": "${modid}:items/${data.texture}"
+            "layer0": "${modid}:items/${(item??)?then(item, data).texture}"
         </#if>
     }
+    <#if data?? && data.getModElement().getTypeString() == "item" && data.filterModels()?has_content>,
+    "overrides": [
+        <#list data.filterModels().entrySet() as model>
+        {
+            "predicate": {
+            <#list model.getKey().split(",") as state>
+                <#assign prop = generator.map(state.split("=")[0], "itemproperties")>
+                "${prop}": ${(state.split("=")[1]?number?float * 1000)?int / 1000}<#sep>,

```

Comment:
Why is this parsing applied here? Does Minecraft require precision up to 1000?

Reply:
I put the step size on the property `JSpinner` to <kbd>0.001</kbd> and I didn't like the way it was stored because it could be then set to e.g. <kbd>0.000993782<...></kbd> instead of <kbd>0.001</kbd>.

Reply:
Check clock.json, it uses very precise values

Reply:
How much decimal digits is that, though? Do I put 6 or 9?

Reply:
How about passing the raw float value as stored in the state map?

Reply:
9 should be fine, one can type 0.1 if they want to still